chain/chain/src/chain.rs

```diff
@@ -1847,7 +1848,8 @@ impl Chain {
     pub fn get_final_transaction_result(
         &mut self,
         hash: &CryptoHash,
-    ) -> Result<FinalExecutionOutcomeView, Error> {
+        fetch_receipt: bool,
```

Comment:
It seems that we don't need to have a flag here, but instead, we need a helper function receiving `FinalExecutionOutcome` and producing `Vec<Receipt>`.